hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -230,22 +231,27 @@ public AzureIngressHandler getIngressHandler() {
    */
   private AzureIngressHandler createIngressHandler(AbfsServiceType serviceType,
       DataBlocks.BlockFactory blockFactory,
-      int bufferSize, boolean isSwitch) throws IOException {
+      int bufferSize, boolean isSwitch, AzureBlockManager blockManager) throws IOException {
     lock.lock();
     try {
+      this.client = clientHandler.getClient(serviceType);
       if (switchCompleted && ingressHandler != null) {
         return ingressHandler; // Return the existing ingress handler
       }
-      this.client = clientHandler.getClient(serviceType);
-      if (serviceType == AbfsServiceType.BLOB) {
+      if (isDFSToBlobFallbackEnabled && serviceTypeAtInit == AbfsServiceType.BLOB) {
```

Comment:
This happens already because we don't have switchHandler handling in case of DFS appends are they don't throw the InvalidIngressServiceException, and we switch only if that exception is caught. So DFSIngressHandler to BlobIngress is not posssible via code